src/clinical/id-generator.ts

```diff
@@ -0,0 +1,293 @@
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import axios from 'axios';
+import memoize from 'memoizee';
+import ms from 'ms';
+import {
+	Biomarker,
+	ClinicalInfo,
+	Comorbidity,
+	CompletionStats,
+	Donor,
+	Exposure,
+	FamilyHistory,
+	FollowUp,
+	PrimaryDiagnosis,
+	SchemaMetadata,
+	Specimen,
+	Treatment,
+} from './clinical-entities';
+import { ClinicalEntitySchemaNames } from '../common-model/entities';
+import { config } from '../config';
+
+interface IdGenerationRequest {
+	programId: string;
+	submitterId: any;
+	submitterDonorId: string;
+	testInterval: string;
+	family_relative_id: string;
+	comorbidityTypeCode: string;
+	entityType: string;
+}
+
+const request = {
+	programId: 'null',
+	submitterId: 'null',
+	submitterDonorId: 'null',
+	testInterval: 'null',
+	family_relative_id: 'null',
+	comorbidityTypeCode: 'null',
+	entityType: 'null',
+};
+
+export interface PartialDonor {
+	_id?: string;
+	__v?: number; // mongodb property not being filtered out
+	createBy?: string;
+	schemaMetadata?: SchemaMetadata | undefined;
+	donorId?: number;
+	gender?: string;
+	submitterId?: string;
+	programId?: string;
+	specimens?: Array<Specimen> | undefined;
+	clinicalInfo?: ClinicalInfo;
+	primaryDiagnoses?: Array<PrimaryDiagnosis>;
+	familyHistory?: Array<FamilyHistory>;
+	comorbidity?: Array<Comorbidity>;
+	followUps?: Array<FollowUp>;
+	treatments?: Array<Treatment>;
+	exposure?: Array<Exposure>;
+	biomarker?: Array<Biomarker>;
+	createdAt?: string;
+	updatedAt?: string;
+	completionStats?: CompletionStats;
+}
+
+export async function getId(req: IdGenerationRequest) {
+	console.log('getId function called');
+	const token = await getToken();
+	const headers = {
+		headers: {
+			Authorization: 'Bearer ' + token,
+		},
+	};
+
+	try {
+		const response = await axios.get(
+			// `http://localhost:9001/${req.programId}/${req.submitterId}/${req.submitterDonorId}/${req.testInterval}/${req.family_relative_id}/${req.comorbidityTypeCode}/${req.entityType}`,
+			config.getConfig().tokenUrl() +
+				`${req.programId}/${req.submitterId}/${req.submitterDonorId}/${req.testInterval}/${req.family_relative_id}/${req.comorbidityTypeCode}/${req.entityType}`,
+			headers,
+		);
+		console.log('getId response: ' + response.data.entityId + ' - ' + response.data.entityType);
```

Comment:
Just FYI string concatenation like this is generally out of favour compared to template literals
```
`getId response: ${response.data.entityId} - ${response.data.entityType}`
```

String + value (concatenation) can sometimes have unexpected results when parsing a value that is not originally a string
Template literals are also _slightly_ cleaner to read + write

Reply:
have removed all unwanted console logs

Reply:
Dan's suggestion, for clarity
```suggestion
		console.log(`getId response: ${response.data.entityId} - ${response.data.entityType}`);
```